src/us/kbase/common/utils/CondorUtils.java

```diff
@@ -257,82 +176,72 @@ public static boolean checkJobIsDoneWithoutError(String jobId) {
-		IntAndStatus jobIdAndStatus = schedd.newJob(transaction, clusterId);
-		int jobId = jobIdAndStatus.getInteger();
-		
-		// Build the Job's ClassAd.
-		ClassAdStructAttr[] jobAd = buildJobAd(owner, jobFileLocation, clusterId, jobId);	
-		
-		// Submit the Job's ClassAd.
-		schedd.submit(transaction, clusterId, jobId, jobAd);
-		
-		
-		// Commit the transaction.
-		schedd.commitTransaction(transaction);
-	
-		// Ask the Schedd to kick off the Job immediately.
-		schedd.requestReschedule();
-		
-	    return jobId;    	
-	}
-	*/
+ 
 	
 	
-	public static int submitToCondorCLI ( String submitFilePath ) throws IOException
+	public static float submitToCondorCLI ( String ujsJobId, String selfExternalUrl, String submitFilePath, String aweClientGroups ) throws IOException
 	{
-		int jobId = 0;
+		float jobId = 0;
 		String line = "";
+		int exitVal = 0;
+
 		Runtime r = Runtime.getRuntime();
-		
-		// XXX: Hardcoded path to the script dir:
-		// cd to the scripts directory and purge out temp.sub
-		// Process p = r.exec( "cd /home/submitter/submit/njs_wrapper/scripts" );
-		// r.exec( "rm -f temp.sub" );
-		
-		// Build a file in njs_wrapper/scripts directory named temp.sh
-		// TODO: will later become the 'executable' attribute inside the submit file 
-		// p = r.exec( "echo " + submitFilePath + " > temp.sh");
 
+		// TODO: Change path to condor_submit script to a relative path: like ../scripts/condor_submit.sh
 		String[] cmdScript = new String[]{ "/bin/bash", "/home/submitter/submit/njs_wrapper/scripts/condor_submit.sh",
+				ujsJobId,
 				submitFilePath };
 		
-		// Execute job submit script with temp.sub as the submit fle:
+		// Execute job submit script with submitFilePath as the submit fle path:
 		Process p = r.exec( cmdScript );
 		
+		try {
+			p.waitFor();
+		} catch (InterruptedException e) {			
+			e.printStackTrace();
+		}
+		/*
 		BufferedReader b = new BufferedReader( new InputStreamReader( p.getInputStream() ) );
-		while ((line = b.readLine()) != null) {
+		
+		line = b.readLine();
+		System.out.println(line);		
+		while (( line = b.readLine()) != null ) {
 		  System.out.println(line);
+		  if( line.contains( "** Proc" ) ) break;
 		}
-		b.close();		
+		b.close();
 		
+		if( line == null || !line.contains( "** Proc" ) ) {
+			System.err.println( "ERROR ERROR ERROR: CondorUtils::submitToCondorCLI: Could not parse jobId from condor_submit IO" );			
+            throw new IOException(  "CondorUtils::submitToCondorCLI: Could not parse jobId from condor_submit IO" ); 			
+		}
+		*/
+		exitVal = p.exitValue();		
+		if ( exitVal == 0 ) { // success
+			
+			// Determine job id (cluster id dot zero) of the job just submitted:			
+	        // parse the substring after 'c' for proc from line
+			// TODO: Don't parse... FORCE!!! Force a 'batch' job id (coordianted with ujs/catalog)
+			//     Back up in the call stack heirarchy (add switch to condor_submit.sh to force a 'batch' job id)
+			/*
+			try{
```

Comment:
should this block be uncommented? otherwise I think returned jobID will always be 0.0?